src/v/cloud_topics/level_one/frontend_reader/level_one_reader.cc

```diff
@@ -51,48 +48,79 @@ level_one_log_reader_impl::level_one_log_reader_impl(
     vlog(_log.debug, "New reader created {}", _config);
 }
 
+/*
+ * Error handling
+ * ==============
+ *
+ * Exceptions should not be used unless you intend for the exception to be
+ * propogated back to the user of a model::record_batch_reader. In this case an
+ * exception that carries a string message can be useful for debugging.
+ */
 ss::future<model::record_batch_reader::storage_t>
 level_one_log_reader_impl::do_load_slice(
   model::timeout_clock::time_point deadline) {
-    chunked_circular_buffer<model::record_batch> res;
-
-    // First switch: ensure batches are materialized or the reader
-    // reaches end-of-stream.
-    switch (_state) {
-    case state::empty:
-        co_await fetch_metadata(deadline);
-        [[fallthrough]];
-    case state::ready:
-        co_await materialize_batches(deadline);
-        [[fallthrough]];
-    case state::materialized:
-    case state::end_of_stream:
-        // Handled in the next switch statement.
-        break;
-    }
+    try {
+        // First switch: ensure batches are materialized or the reader
+        // reaches end-of-stream.
```

Comment:
Very much appreciate that now all the updates to _state are in this method, thank you!